ntp-daemon/src/config/peer.rs

```diff
@@ -16,14 +16,52 @@ impl Default for PeerHostMode {
     }
 }
 
-#[derive(Debug, PartialEq, Eq, Clone)]
-pub struct PeerConfig {
+#[derive(Deserialize, Debug)]
+pub enum PeerType {
+    Standard,
+    Pool,
+}
+
+#[derive(Deserialize, Debug, PartialEq, Eq, Clone)]
+pub struct StandardPeerConfig {
     // We ensure that this is an address with a host and port part
     // however the host may or may not be valid.
     pub addr: String,
     pub mode: PeerHostMode,
 }
 
+#[derive(Deserialize, Debug, PartialEq, Eq, Clone)]
+pub struct PoolPeerConfig {
+    // We ensure that this is an address with a host and port part
+    // however the host may or may not be valid.
+    pub addr: String,
+    pub mode: PeerHostMode,
+    pub max_peers: usize,
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum PeerConfig {
+    Standard(StandardPeerConfig),
+    Pool(PoolPeerConfig),
+    // Consul(ConsulPeerConfig),
+}
+
+impl PeerConfig {
+    pub fn addr(&self) -> &str {
```

Comment:
Are you sure we want to add these kinds of methods on PeerConfig? It looks to me like they quickly become irrelevant for other types of configurations. Instead maybe match on the PeerConfig variant (and add a little `unimplemented!` for variants we don't support right now)?